### PR TITLE
Use manifest path set for andorid_library rules for project generation

### PR DESCRIPTION
**Pull request by kageiit**
Also fallback to the default manifest path set in the `intellij` buckconfig section

### CODE REVIEW

**Comment by facebook-github-bot-4** [comment]
By analyzing the blame information on this pull request, we identified @alsutton, @tonycosentini and @andrewjcg to be potential reviewers.

**Comment by facebook-github-bot-0** [comment]
@kageiit updated the pull request - [view changes](https://github.com/facebook/buck/pull/892/files/77b422050f16965e5d5b6439c5ef2de8f7acc2e1..8fc97d61c2f0375da16f8eb0ae4c998d889037a9)

File: src/com/facebook/buck/jvm/java/intellij/IjProject.java [hunk 1]
```diff
@@ -146,6 +148,16 @@ public Path getAndroidManifestPath(TargetNode<AndroidBinaryDescription.Arg> targ
           }
 
           @Override
+          public Optional<Path> getLibraryAndroidManifestPath(TargetNode<AndroidLibraryDescription.Arg> targetNode) {
+            Optional<SourcePath> manifestPath = targetNode.getConstructorArg().manifest;
+            if (manifestPath.isPresent()) {
```
Comment (marcinkosiba) [inline src/com/facebook/buck/jvm/java/intellij/IjProject.java]: `return manifestPath.transform(sourcePathResolver.getAbsolutePathFunction()).or( intellijConfig.getAndroidManifest());`

**Comment by kageiit** [comment]
Sounds good. I will address your comment and add integration tests

**Comment by kageiit** [comment]
@marcinkosiba the `android_project_with_gradle_conventions` integration test fixture covers this case already

**Comment by kageiit** [comment]
@marcinkosiba I added a test for the default manifest path by deleting the manifest path from one of the test fixtures and letting it fallback to the value specified in buckconfig

**Comment by kageiit** [comment]
@marcinkosiba @dreiss ping

**Pull request closed without merging**